File: Paramount.Betterclassifieds.Business/Events/EventOrganiser.cs
﻿using System;

namespace Paramount.Betterclassifieds.Business.Events
{
    public class EventOrganiser
    {
        public int EventOrganiserId { get; set; }
        public int EventId { get; set; }
        public EventModel Event { get; set; }
        public string UserId { get; set; }
        public string Email { get; set; }
        public Guid? InviteToken { get; set; }
        public string LastModifiedBy { get; set; }
        public DateTime LastModifiedDateUtc { get; set; }
        public DateTime LastModifiedDate { get; set; }
        public bool IsActive { get; set; }
        public bool? SubscribeToPurchaseNotifications { get; set; }
        public bool? SubscribeToDailyNotifications { get; set; }
        
        /// <summary>
        /// Returns whether this organiser is invited
        /// </summary>
        /// <returns></returns>
        public bool IsInvited()
        {
            return UserId.IsNullOrEmpty() && InviteToken.HasValue;
        }
    }

    public enum OrganiserConfirmationResult
    {
        NotFound,
        Success,
        AlreadyActivated,
        MismatchedEmail
    }

    public enum EmailSubscription
    {
        None,
        PerTicketPurchase,
        PerDay
    }
}

File: Paramount.Betterclassifieds.Presentation/Scripts/app/events/organisers/event-organiser.js
(function ($, $p, ko, moment) {

    $p.models.EventOrganiser = function (data) {

        var me = this;

        me.eventOrganiserId = ko.observable(data.eventOrganiserId);
        me.email = ko.observable(data.email);
        me.lastModifiedDate = moment.utc(data.lastModifiedDateUtc).local().format($p.jsToDisplayDateFormat);
        me.status = ko.observable(getStatus(data));
        me.userId = ko.observable(data.userId);
        me.inviteToken = ko.observable(data.inviteToken);

        me.statusClass = ko.computed(function () {
            console.log(me.status());
            if (me.status() === 'Active') {
                return "label-success";
            }

            return 'label-warning';
        });
    }

    function getStatus(organiser) {
        if (organiser.userId !== null && organiser.userId !== '') {
            return "Active";
        }

        if (organiser.inviteToken !== null) {
            return "Invited";
        }

        return null;
    }

})(jQuery, $paramount, ko, moment);